azurerm/resource_arm_data_factory.go

```diff
@@ -0,0 +1,371 @@
+
+	name := d.Get("name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	resourceGroup := d.Get("resource_group_name").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Data Factory %q (Resource Group %q): %s", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_data_factory", *existing.ID)
+		}
+	}
+
+	dataFactory := datafactory.Factory{
+		Location: &location,
+		Tags:     expandTags(tags),
+	}
+
+	if v, ok := d.GetOk("identity.0.type"); ok {
+		identityType := v.(string)
+		dataFactory.Identity = &datafactory.FactoryIdentity{
+			Type: &identityType,
+		}
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, name, dataFactory, ""); err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return err
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Data Factory %s (resource group %s) ID", name, resourceGroup)
+	}
+
+	if hasRepo, repo := expandArmDataFactoryRepoConfiguration(d); hasRepo {
+		repoUpdate := datafactory.FactoryRepoUpdate{
+			FactoryResourceID: resp.ID,
+			RepoConfiguration: repo,
+		}
+		if _, err = client.ConfigureFactoryRepo(ctx, location, repoUpdate); err != nil {
+			return err
+		}
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmDataFactoryRead(d, meta)
+}
+
+func resourceArmDataFactoryRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataFactoryClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["factories"]
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error making Read request on AzureRM Data Factory %s: %+v", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	repoType, repo := flattenArmDataFactoryRepoConfiguration(&resp)
+	if repoType == datafactory.TypeFactoryVSTSConfiguration {
+		if err := d.Set("vsts_configuration", repo); err != nil {
+			return fmt.Errorf("Error setting Data Factory %q `vsts_configuration` (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+	if repoType == datafactory.TypeFactoryGitHubConfiguration {
+		if err := d.Set("github_configuration", repo); err != nil {
+			return fmt.Errorf("Error setting Data Factory %q `github_configuration` (Resource Group %q): %+v", name, resourceGroup, err)
```

Comment:
for consistency with the other resources can we make this:

```suggestion
			return fmt.Errorf("Error setting `github_configuration`: %+v", err)
```